.github/workflows/commitlint.yml

```diff
@@ -0,0 +1,26 @@
+name: Commitlint
+on:
+  push:
+    tags:
+      - v*
+    branches:
+      - master
+      - release-*
+  pull_request:
+    branches:
+      - master
+      - release-*
+
+jobs:
+  lint:
+    runs-on: ubuntu-18.04
+    env:
+      GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```

Comment:
> There is no need for this one. See checkout docs: https://github.com/actions/checkout In particular: `Only a single commit is fetched by default, for the ref/SHA that triggered the workflow. Set fetch-depth: 0 to fetch all history for all branches and tags`

removed the depth, thanks @synarete

Reply:
but this seems to have rendered the workflow definition invalid. 🤷‍♂️

Reply:
@obnoxxx My bad; doc states explicitly that `fetch-depth: 0` is needed: https://github.com/wagoid/commitlint-github-action#usage

Reply:
```Note: It's necessary that you specify the fetch-depth argument to actions/checkout@v2 step. By default they fetch only latest commit of the branch, but we need more commits since we validate a range of commit messages.```

Reply:
> `Note: It's necessary that you specify the fetch-depth argument to actions/checkout@v2 step. By default they fetch only latest commit of the branch, but we need more commits since we validate a range of commit messages.`

@synarete thanks - I re-added it and now it works again 😄